stdlib/LinearAlgebra/test/blas.jl

```diff
@@ -78,6 +78,28 @@ Random.seed!(100)
                 @test BLAS.iamax(z) == argmax(map(x -> abs(real(x)) + abs(imag(x)), z))
             end
         end
+        @testset "rot!" begin
+            if elty <: Real
+                x = convert(Vector{elty}, randn(n))
+                y = convert(Vector{elty}, randn(n))
+                c = rand(elty)
+                s = rand(elty)
+                x2, y2 = BLAS.rot!(n,copy(x),1,copy(y),1,c,s)
+                @test x2 ≈ c*x + s*y
+                @test y2 ≈ -s*x + c*y
```

Comment:
I'd suggest to include testing that `x` and `y` are actually overwritten, something along the lines
```suggestion
                x2 = copy(x)
                y2 = copy(y)
                BLAS.rot!(n, x, 1, y, 1, c, s)
                @test x ≈ c*x2 + s*y2
                @test y ≈ -s*x2 + c*y2
```
and similarly below.